scripts/azure_kernel_validation/jenkins-azure.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+set -xe
+
+validate() {
+    BASEDIR=$1
+    BUILD_NAME=$2
+    BUILD_NUMBER=$3
+    USERNAME=$4
+    PASSWORD=$5
+    SMB_SHARE_URL=$6
+    
+    KERNEL_VERSION_FILE="./kernel_version${BUILD_NUMBER}/scripts/package_building/kernel_versions.ini"
+    KERNEL_FOLDER=$(crudini --get $KERNEL_VERSION_FILE KERNEL_BUILT folder)
+    DESIRED_KERNEL_VERSION=$(crudini --get $KERNEL_VERSION_FILE KERNEL_BUILT version)
+    DESIRED_KERNEL_TAG=$(crudini --get $KERNEL_VERSION_FILE KERNEL_BUILT git_tag)
+    
+    pushd "$BASEDIR/scripts/azure_kernel_validation"
+    bash create_azure_vm.sh --build_number "$BUILD_NAME$BUILD_NUMBER" --clone_repo y \
+                            --vm_params "username=$USERNAME,password=$PASSWORD,samba_path=$SMB_SHARE_URL/temp-kernel-artifacts,kernel_path=$KERNEL_FOLDER" \
+                            --deploy_data azure_kernel_validation --resource_group kernel-validation \
+                            --os_type $OS_TYPE
+    popd
+    
+    . "$BASEDIR/scripts/package_building/utils.sh"
+    
+    INTERVAL=5
+    COUNTER=0
+    
+    while [ $COUNTER -lt $AZURE_MAX_RETRIES ]; do
+        public_ip_raw=$(az network public-ip show --name "$BUILD_NAME$BUILD_NUMBER-PublicIP" --resource-group kernel-validation --query \'{address: ipAddress }\')
+        public_ip=`echo $public_ip_raw | awk \'{if (NR == 1) {print $3}}\' | tr -d \'"\'`
+        if [ !  -z $public_ip ]; then
+            echo "Public ip available: $public_ip."
+            break
+        else
+            echo "Public ip not available."
+        fi
+        let COUNTER=COUNTER+1
+    
+        if [ -n "$INTERVAL" ]; then
+            sleep $INTERVAL
+        fi
+    done
+    
+    if [ $COUNTER -eq $AZURE_MAX_RETRIES ]; then
+        echo "Failed to get public ip. Exiting..."
+        exit 2
+    fi
+    
+    INTERVAL=5
+    COUNTER=0
+    while [ $COUNTER -lt $AZURE_MAX_RETRIES ]; do
+        KERNEL_NAME=`ssh -i ~/azure_priv_key.pem -o StrictHostKeyChecking=no -o ConnectTimeout=10 ubuntu@$public_ip uname -r || true`
```

Comment:
Parameterize PRIVATE_KEY_PATH="~/azure_priv_key.pem", VM_USER_NAME="ubuntu" (these are the defaults)